derrida/books/management/commands/instance_data.py

```diff
@@ -0,0 +1,122 @@
+
+    def handle(self, *args, **kwargs):
+        base_filename = 'derrida-instance-data'
+        if kwargs['directory']:
+            base_filename = os.path.join(kwargs['directory'], base_filename)
+
+        instancedata = [self.instance_data(instance) for instance in Instance.objects.filter(cited_in__isnull=False)]
+
+        # list of dictionaries can be output as is for JSON export
+        with open('{}.json'.format(base_filename), 'w') as jsonfile:
+            json_instancedata = [{field: inst[field] for field in inst.keys() if inst[field] not in [None, '', []]} for inst in instancedata]
+            json.dump(json_instancedata, jsonfile, indent=2)
+
+        # generate CSV export
+        with open('{}.csv'.format(base_filename), 'w') as csvfile:
+            # write utf-8 byte order mark at the beginning of the file
+            csvfile.write(codecs.BOM_UTF8.decode())
+
+            csvwriter = csv.DictWriter(csvfile, fieldnames=self.csv_fields)
+            csvwriter.writeheader()
+
+            for instance in instancedata:
+                csvwriter.writerow(self.flatten_dict(instance))
+
+    def instance_data(self, instance):
+        '''Generate a dictionary of data to export for a single
+         :class:`~derrida.books.models.Instance` object'''
+        return OrderedDict([
+            ('id', instance.get_uri()),
+            ('item_type', instance.item_type),
+            ('work_title', instance.work.primary_title),
+            ('work_short_title', instance.work.short_title),
+            ('alternate_title', instance.alternate_title),
+            ('work_year', instance.work.year),
+            ('copyright_year', instance.copyright_year),
+            # TODO: take into account print date day/month/year known fields
+            # ('print_date', str(instance.print_date) if instance.print_date else ''),
+            ('work_authors', [str(author) for author in instance.work.authors.all()]),
+            ('publisher', instance.publisher.name if instance.publisher else ''),
+            ('pub_place', [place.name for place in instance.pub_place.all()]),
+            ('is_extant', instance.is_extant),
+            ('is_annotated', instance.is_annotated),
+            ('is_translation', instance.is_translation),
+            ('has_dedication', instance.has_dedication),
+            ('has_insertions', instance.has_insertions),
+            ('copy', instance.copy),
+            ('dimensions', instance.dimensions),
+            ('work_uri', instance.work.uri),
+            ('work_subjects', [str(subject) for subject in instance.work.subjects.all()]),
+            ('languages', [str(language) for language in instance.languages.all()]),
+            ('journal_title', instance.journal.name if instance.journal else ''),
+            ('book_title', instance.collected_in.display_title() if instance.collected_in else ''),
+            ('book_title_uri', instance.collected_in.get_uri() if instance.collected_in else ''),
```

Comment:
- [ ] Double check that this URI appears in `id` column

Reply:
There is only one URI that doesn't exist in the `id` column:

https://derridas-margins.princeton.edu/titles/151/

@rlskoeser we were concerned about missing many titles, correct? Any clue why this one URL would be missing?

Reply:
I checked the notes and found this:
> Stub collection work Trends in European and American linguistics 1930-1960 created from book section for Glossematics

It looks like we have one reference linked to the book section and one linked to the book. We could correct that (they're both on the same page) — or maybe even delete one? (it looks like the exact same content to me). However, we discussed including books that collect any book sections that are in scope; if we include them, that will solve this too.

Reply:
I feel like deleting is the most straightforward, and then if we want to implement that feature we can later? Let me know

Reply:
@kmcelwee after thinking about it more, I'd rather not touch the data — making any data edits is a slippery slope, because I'm pretty sure there are other duplicates as well, but I don't think we should be fixing them at this point.

We need to include books that include book sections anyway, for other reasons. I can help with the queryset filter if you aren't sure the best way to construct it.

Reply:
See 6ff6ba9 and https://github.com/Princeton-CDH/derrida-django/pull/274#discussion_r707582619